src/main/java/tc/oc/pgm/modules/ModifyBowProjectileMatchModule.java

```diff
@@ -40,44 +41,52 @@ public ModifyBowProjectileMatchModule(
 
   @EventHandler(ignoreCancelled = true)
   public void changeBowProjectile(EntityShootBowEvent event) {
-    Plugin plugin = PGM.get();
-    Entity newProjectile;
-
-    if (this.cls == Arrow.class && event.getProjectile() instanceof Arrow) {
-      // Don't change the projectile if it's an Arrow and the custom entity type is also Arrow
-      newProjectile = event.getProjectile();
-    } else {
-      // Replace the projectile
-      World world = event.getEntity().getWorld();
+    if (event.getProjectile() instanceof Arrow) {
       Projectile oldEntity = (Projectile) event.getProjectile();
-      newProjectile = world.spawn(oldEntity.getLocation(), this.cls);
-      event.setProjectile(newProjectile);
 
-      // Copy some things from the old projectile
-      newProjectile.setVelocity(oldEntity.getVelocity());
-      newProjectile.setFallDistance(oldEntity.getFallDistance());
-      newProjectile.setFireTicks(oldEntity.getFireTicks());
-
-      if (newProjectile instanceof Projectile) {
-        ((Projectile) newProjectile).setShooter(oldEntity.getShooter());
-        ((Projectile) newProjectile).setBounce(oldEntity.doesBounce());
+      // EnderPearl isn't an entity type in world#spawn so this is a workaround
+      if (EnderPearl.class.isAssignableFrom(this.cls)) {
+        event.getEntity().launchProjectile(EnderPearl.class).setVelocity(oldEntity.getVelocity());
+        event.setCancelled(true);
+        return;
```

Comment:
This workarround should still follow the logic of the function, not return straight away without setting shoother etc.
Just replace `world.spawn(..)` with the `entity.launchProjectile(...)` method, and let other things be handled the same

Reply:
good catch